app/views/incomeSources/add/IncomeSourceCheckDetails.scala.html

```diff
@@ -0,0 +1,208 @@
+            controllers.incomeSources.add.routes.AddBusinessTradeController.show(isAgent = false, isChange = true).url
+        }
+    }
+
+    @changeBusinessAddressURL = @{
+        if(isAgent) {
+            controllers.incomeSources.add.routes.AddBusinessAddressController.showAgent(isChange = true).url
+        } else {
+            controllers.incomeSources.add.routes.AddBusinessAddressController.show(isChange = true).url
+        }
+    }
+
+    @getLongAddressFromBusinessAddressDetails(address1: String, address2: Option[String], address3: Option[String], address4: Option[String], postCode: Option[String], countryCode: Option[String]) = @{
+        List(Some(address1),
+            address2,
+            address3,
+            address4,
+            postCode,
+            countryCode
+        ).flatten.filter(_.nonEmpty).mkString("<br>")
+    }
+
+
+    @changeStartDateURL = @{
+        controllers.incomeSources.add.routes.AddIncomeSourceStartDateController.show(isAgent = isAgent, isChange = true, viewModel.incomeSourceType).url
+    }
+
+    @changeAccountingMethodURL = @{
+        if(isAgent) {
+            controllers.incomeSources.add.routes.IncomeSourcesAccountingMethodController.changeIncomeSourcesAccountingMethodAgent(viewModel.incomeSourceType).url
+        } else {
+            controllers.incomeSources.add.routes.IncomeSourcesAccountingMethodController.changeIncomeSourcesAccountingMethod(viewModel.incomeSourceType).url
+        }
+    }
+
+@mainTemplate(
+    pageTitle = getMessage("title"),
+    backUrl = Some(backUrl),
+    useFallbackBackLink = true,
+    isAgent = isAgent,
+    btaNavPartial = user.btaNavPartial) {
+
+    @hmrcPageHeading(PageHeading(
+        text = getMessage("title"),
+        headingClasses = Some("govuk-heading-l"),
+        section = Some(getMessage("heading")),
+        captionClasses = Some("govuk-caption-l")
+    ))
+
+    @formWithCSRF(action = postAction) {
+
+        <dl class="govuk-summary-list" id="check-details-table">
+            @if(viewModel.incomeSourceType == SelfEmployment){
+                <div class="govuk-summary-list__row">
+                    <dt class="govuk-summary-list__key">@getMessage("business-name")</dt>
+                    <dd class="govuk-summary-list__value" id="business-name-value">@{
+                        viewModel.businessName.getOrElse("")
+                    }</dd>
+                    <dd class="govuk-summary-list__actions govuk-!-text-align-right govuk-!-margin-0">
+                        <a class="govuk-link" href=@changeBusinessNameURL>
+                        @getMessage("change")
+                        </a>
+                    </dd>
+                </div>
+            }
+            <div class="govuk-summary-list__row">
+                <dt class="govuk-summary-list__key">@getMessage("start-date")</dt>
+                <dd class="govuk-summary-list__value" id="start-date-value">@{
+                    viewModel.businessStartDate.get.toLongDate
```

Comment:
if we can avoid calling .get and .head in the view? do we have a default value or we should raise an exception when constructing model ? it seem that we this field should not be Optional;

if this is different across view (entities: in some Optional and in some mandatory ) we would have to use different model definition for this